kong/plugins/grpc-gateway/deco.lua

```diff
@@ -199,6 +200,34 @@ local function unframe(body)
   return body:sub(pos, frame_end), body:sub(frame_end + 1)
 end
 
+--[[
+  // Set value `v` at `path` in table `t`
+  // Path contains value address in dot-syntax. For example:
+  // path="a.b.c" would lead to `t[a][b][c] = c`
+]]
+local function add_to_table( t, path, v )
+  local function patch_table( t, path, v )
+    local k = table.remove( path, 1 ) -- pop out first key
+
+    if #path > 0 then
```

Comment:
on an array where we know there are no holes, a faster "emptiness" test is just `if path[1] ~= nil then...`  (or `== nil` and invert the `then` and `else` clauses, whichever is more readable)